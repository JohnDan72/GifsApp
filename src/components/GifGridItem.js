
import React from 'react';
import PropTypes from 'prop-types';
import { Panel } from 'rsuite';
import styles from '../css/GifGridItem.module.css';


const propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    shortName: PropTypes.string.isRequired,
}

const GifGridItem = ({ id, title, url , shortName }) => {
    return (
        <Panel bodyFill className={styles.rsPanel}> 
            <img src={url} className={styles.imgCard} alt={title} />
            <Panel header={shortName}></Panel>
        </Panel>
    );
}

GifGridItem.propTypes = propTypes;
// #endregion

export default GifGridItem;