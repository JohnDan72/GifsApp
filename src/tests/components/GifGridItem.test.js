import { shallow } from 'enzyme';
import React from 'react';
import GifGridItem from '../../components/GifGridItem';



describe('GifGridItem Test' , () => {
    const [ title , url , shortName ] = [   
        'Saitama One Punch Man',
        'https://media1.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=3828c31fn23sgcds2q8sw1ylwqi6953rmkhs4xs1fbyyxlwv&rid=giphy.gif&ct=g',
        'Saitama One Punch Man'
    ];

    let wrapper = shallow( <GifGridItem title={title} url={url} shortName={shortName} /> );

    beforeEach(() => {
        wrapper = shallow( <GifGridItem title={title} url={url} shortName={shortName} /> );
    });

    test('should render component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should has an alt img property equals to the title props', () => {
        const imgProps = wrapper.find('img').props();
        // console.log(imgProps)
        const altProperty = imgProps.alt;
        const srcAttribute = imgProps.src;

        // console.log(altProperty);
        expect(altProperty).toBe(title);
        expect(srcAttribute).toBe(url);

    });

    test('should has a header prop within Panel component and be equals to title', () => {
        const panelComponent = wrapper.find('Panel').at(1);
        const header = panelComponent.props().header;
        // console.log(panelComponent.props() )

        expect(header).toBe(title);
    });

    test('should has rsPanel className in styles og First Palen component', () => {
        const classNameExpected = 'rsPanel';

        const panelComponent = wrapper.find('Panel').at(0);
        const className = panelComponent.props().className;
        // console.log(panelComponent.prop('className') )

        expect(className.includes(classNameExpected)).toBe(true);
    })
    
    
})