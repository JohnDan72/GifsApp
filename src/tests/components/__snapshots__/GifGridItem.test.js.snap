// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem Test should render component 1`] = `
<Panel
  bodyFill={true}
  className="rsPanel"
>
  <img
    alt="Saitama One Punch Man"
    className="imgCard"
    src="https://media1.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=3828c31fn23sgcds2q8sw1ylwqi6953rmkhs4xs1fbyyxlwv&rid=giphy.gif&ct=g"
  />
  <Panel
    header="Saitama One Punch Man"
  />
</Panel>
`;
